tests/unit_tests/constraint/test_skeleton.py

```diff
@@ -0,0 +1,119 @@
+import networkx as nx
+import numpy as np
+import pandas as pd
+import pytest
+
+from dodiscover import Context
+from dodiscover.ci import GSquareCITest, Oracle
+from dodiscover.constraint import LearnSkeleton
+from dodiscover.constraint.utils import dummy_sample
+from tests.unit_tests.ci.testdata import bin_data, dis_data
+
+
+def common_cause_and_collider():
+    """Test orienting a common cause and a collider.
+
+    The following graph has some complexities to test the PC algorithm
+    with the Oracle setting: ``1 <- 0 -> 2 <- 3``.
+    """
+    # build initial DAG
+    ed1, ed2 = ({}, {})
+    incoming_graph_data = {0: {1: ed1, 2: ed2}, 3: {2: ed2}}
+    G = nx.DiGraph(incoming_graph_data)
+    return G
+
+
+def collider():
+    # build initial DAG
+    edge_list = [
+        ("x", "y"),
+        ("z", "y"),
+    ]
+    G = nx.DiGraph(edge_list)
+    return G
+
+
+def complex_graph():
+    """Complex graph from Figure 2 of :footcite:`Colombo2012`.
+
+    Does not include the latent edges.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    edge_list = [
+        ("x4", "x1"),
+        ("x2", "x5"),
+        ("x3", "x2"),
+        ("x3", "x4"),
+        ("x2", "x6"),
+        ("x3", "x6"),
+        ("x4", "x6"),
+        ("x5", "x6"),
+    ]
+    G = nx.DiGraph(edge_list)
+    return G
+
+
+@pytest.mark.parametrize(
+    ("indep_test_func", "data_matrix", "g_answer"),
+    [
+        (
+            GSquareCITest(),
+            np.array(bin_data).reshape((5000, 5)),
+            nx.DiGraph(
+                {
+                    0: (1,),
+                    1: (),
+                    2: (3, 4),
+                    3: (1, 2),
+                    4: (1, 2),
+                }
+            ),
+        ),
+        (
+            GSquareCITest("discrete"),
+            np.array(dis_data).reshape((10000, 5)),
+            nx.DiGraph(
+                {
+                    0: (2,),
+                    1: (2, 3),
+                    2: (),
+                    3: (),
+                    4: (3,),
+                }
+            ),
+        ),
+    ],
+)
+def test_learn_skeleton_with_data(indep_test_func, data_matrix, g_answer, alpha=0.01):
```

Comment:
Is there a reason for ```alpha = 0.01``` instead of the 'common' ```0.05```?

Reply:
Just cuz the CI test gets the wrong answer at 0.05. I added a comment.